src/ui/overlay.cpp

```diff
@@ -26,9 +26,11 @@ void OverlayImage::render(QPainter *painter, int x, int y) {
 void Overlay::renderItems(QPainter *painter) {
     if (this->hidden) return;
 
+    qreal oldOpacity = painter->opacity();
```

Comment:
There is a `painter->save()` and `painter->restore()` pair that pushes and pops the painter's state.  This is fine, too, though.